File: erp-backend/erp-customer/src/main/java/com/erp/customer/entity/Customer.java
package com.erp.customer.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.erp.common.base.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.util.List;

/**
 * 客户实体
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("customer")
public class Customer extends BaseEntity {

    /**
     * 客户编码
     */
    private String customerCode;

    /**
     * 客户名称
     */
    private String customerName;

    /**
     * 客户类型（1国内 2国外）
     */
    private Integer customerType;

    /**
     * 行业
     */
    private String industry;

    /**
     * 客户等级（1A 2B 3C 4D）
     */
    private Integer customerLevel;

    /**
     * 客户来源
     */
    private String source;

    /**
     * 税号
     */
    private String taxNumber;

    /**
     * 开户行
     */
    private String bankName;

    /**
     * 银行账号
     */
    private String bankAccount;

    /**
     * 账期
     */
    private String paymentTerms;

    /**
     * 信用额度
     */
    private BigDecimal creditLimit;

    /**
     * 地址
     */
    private String address;

    /**
     * 状态（0停用 1启用）
     */
    private Integer status;

    /**
     * 联系人列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<CustomerContact> contacts;
}

File: erp-frontend/src/types/customer.ts
/** 客户 */
export interface Customer {
  id: number;
  customerCode: string;
  customerName: string;
  customerType: number;
  industry: string;
  customerLevel: number;
  source: string;
  taxNumber: string;
  bankName: string;
  bankAccount: string;
  paymentTerms: string;
  creditLimit: number;
  address: string;
  status: number;
  createdAt: string;
  remark: string;
}

/** 客户详情（含联系人+跟进记录） */
export interface CustomerDetail extends Customer {
  contacts: CustomerContact[];
  recentFollows: CustomerFollow[];
}

/** 客户表单 */
export interface CustomerFormData {
  id?: number;
  customerName: string;
  customerType: number;
  industry: string;
  customerLevel: number;
  source: string;
  taxNumber: string;
  bankName: string;
  bankAccount: string;
  paymentTerms: string;
  creditLimit: number;
  address: string;
  remark: string;
}

/** 客户查询参数 */
export interface CustomerQuery {
  keyword?: string;
  customerType?: number;
  customerLevel?: number;
  industry?: string;
  status?: number;
  pageNum?: number;
  pageSize?: number;
}

/** 客户联系人 */
export interface CustomerContact {
  id: number;
  customerId: number;
  contactName: string;
  position: string;
  phone: string;
  email: string;
  isPrimary: number;
  createdAt: string;
}

/** 联系人表单 */
export interface ContactFormData {
  id?: number;
  customerId: number;
  contactName: string;
  position: string;
  phone: string;
  email: string;
  isPrimary: number;
  remark: string;
}

/** 客户跟进记录 */
export interface CustomerFollow {
  id: number;
  customerId: number;
  contactId: number;
  followType: number;
  content: string;
  followTime: string;
  nextFollowTime: string;
  operatorId: number;
  operatorName: string;
  contactName: string;
  createdAt: string;
}

/** 跟进记录表单 */
export interface FollowFormData {
  customerId: number;
  contactId?: number;
  followType: number;
  content: string;
  followTime: string;
  nextFollowTime?: string;
  remark: string;
}

/** 登录请求 */
export interface LoginRequest {
  username: string;
  password: string;
}

/** 登录响应 */
export interface LoginResult {
  accessToken: string;
  refreshToken: string;
  expiresIn: number;
}

/** 用户信息 */
export interface UserInfo {
  id: number;
  username: string;
  nickname: string;
  realName: string;
  email: string;
  phone: string;
  avatar: string;
  roles: string[];
  permissions: string[];
}
